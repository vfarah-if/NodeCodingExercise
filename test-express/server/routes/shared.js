import app from "../app";
import supertest from "supertest";
const request = supertest(app);

export async function getFetch(url) {
    return await request.get(url);
}