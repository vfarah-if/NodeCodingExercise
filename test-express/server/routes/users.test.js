import { getFetch } from "./shared";

describe("users api", () => {
  it("should get the users json response", async (done) => {
    const actualResponse = await getFetch("/users");
    expect(actualResponse).toBeTruthy();
    expect(actualResponse.status).toBe(200);
    expect(actualResponse.text).toMatchSnapshot();
    done();
  });
});
