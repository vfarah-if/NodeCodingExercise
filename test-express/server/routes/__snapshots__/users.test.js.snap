// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users api should get the users json response 1`] = `"[{\\"user\\":{\\"name\\":\\"Jane\\",\\"lastname\\":\\"Doe\\"}},{\\"user\\":{\\"name\\":\\"Jon\\",\\"lastname\\":\\"Doe\\"}},{\\"user\\":{\\"name\\":\\"Michael\\",\\"lastname\\":\\"Caine\\"}},{\\"user\\":{\\"name\\":\\"James\\",\\"lastname\\":\\"Bond\\"}},{\\"user\\":{\\"name\\":\\"Robert\\",\\"lastname\\":\\"Plant\\"}},{\\"user\\":{\\"name\\":\\"Jordan\\",\\"lastname\\":\\"Peterson\\"}}]"`;
